src/charm.py

```diff
@@ -136,10 +143,10 @@ def _process_wheel_file(self, event):  # noqa: C901
                 wheel_data = file.read()
 
                 # Push wheel file to the container and extract it.
-                container.push("/user_provided/wheel_file.whl", wheel_data, make_dirs=True)
+                container.push(f"/user_provided/{filename}", wheel_data, make_dirs=True)
                 container.exec(["apt-get", "update"]).wait()
                 container.exec(["apt-get", "install", "unzip"]).wait()
```

Comment:
Would it not be better to bake the `apt-get` logic directly into the charm (if possible), or put it in the `__init__` function so you don't have to do it every time you update the wheel?

Reply:
I think the wording of this comment is confusing, do you mean put the apt-get logic into the workload container? I.e. Add it into the original docker image? If so, +1 to that, seems like you'll always need unzip so just include it by default.

Reply:
As this charm allows users to deploy with their own image (which should include Python as a minimum), this approach ensures that ```unzip``` is always installed even if it is missing from the image. The logic is included in this method because it ensure that the pebble API is ready, this cannot be guaranteed in the ```__init__``` function (have also tried including it in the ```pebble_ready``` hook, but that did not work either).